src/arkode/arkode_mristep.c

```diff
@@ -1957,20 +3238,25 @@ int mriStep_CheckCoupling(ARKodeMem ark_mem)
     return (ARK_INVALID_TABLE);
   }
 
-  /* check that stage times are sorted */
-  okay = SUNTRUE;
-  for (i = 1; i < step_mem->MRIC->stages; i++)
+  /* check that MRI-GARK stage times are sorted */
+  if ((step_mem->MRIC->type == MRISTEP_IMPLICIT) ||
+      (step_mem->MRIC->type == MRISTEP_EXPLICIT) ||
+      (step_mem->MRIC->type == MRISTEP_IMEX))
```

Comment:
That is already checked [indirectly] by examining the table structure below.

We cannot examine the stagetypes in this routine (at least not currently), because `mriStep_CheckCoupling` is called before the `stage_types` array is allocated/filled.